### PR TITLE
Preprocessing function + Documentation improvement + Percentage delays notebook

### PR DESCRIPTION
**Pull request by nyxgear**
- `preprocess_data_sets(spark, data_sets, parquet_data_sets)` function to uniform preprocessing stage
- `DEV` mode structure for notebooks

**Percentage delays notebook**
- data processing working
- raw data available
- still needed data visualization

**Documentation improvement**
- improved repository's cover README
- solved merge conflicts on documentation, kept parallel:

    - Env and project setup with Conda
    - Env and project setup with Virtualenv

### CODE REVIEW

File: dataset/get_data_sets.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (leonardoarcari) [reply dataset/get_data_sets.sh]: Oh also I see that `get_data_sets.sh` is duplicated both in `dataset` and in `script`.
Keep only the `script` one. Thanks!

File: requirements.txt [hunk 2]
```diff
@@ -1,80 +1,6 @@
...
-statsmodels==0.9.0
-terminado==0.8.1
-testpath==0.3.1
-tornado==5.1
-traitlets==4.3.2
-Twisted==18.7.0
-wcwidth==0.1.7
-webencodings==0.5.1
-widgetsnbextension==3.4.0
-zope.interface==4.5.0
```
[the remark this answers is not in the document]
Comment (nyxgear) [reply requirements.txt]: Yes, tested :) I deleted the whole virtualenv, created and installed only those dependencies you see and it works perfectly.

File: dataset/get_data_sets.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (nyxgear) [reply dataset/get_data_sets.sh]: Yes, actually I just forgot to delete the `./dataset/get_dataset.sh` file. Thanks for the cross check you did. Solved here: https://github.com/polimi-mt-acg/USA-airlines-statistics-1994-2008/pull/4/commits/9ef9d0ceaaba7346f3d98ad6f9ad5ac46de98d98#diff-e2827bf6c618c0f0095a0006e6964b28

I left totally unchanged the `get_dataset.sh` file in the `./script` folder, so it is exactly the one you pasted here above coming from your branch ;) Please, check it again to be sure, but it should be as said.

File: notebooks/utils.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import sys
+from datetime import datetime
+from pathlib import Path
+from pyspark.sql.types import IntegerType
+
+def preprocess_data_sets(spark, data_sets, parquet_data_sets):
```
[the remark this answers is not in the document]
Comment (nyxgear) [reply notebooks/utils.py]: Really good suggestion, I like it!! 
Implemented.
Please have a look at https://github.com/polimi-mt-acg/USA-airlines-statistics-1994-2008/pull/4/files#diff-7ca239fb304fa4e450224cf4ef0d25b5

What do you think about the functions `perform_dataset_preprocessing(spark)` and `load_preprocessed_dataset(spark)` usage now?
https://github.com/polimi-mt-acg/USA-airlines-statistics-1994-2008/blob/0e6463111b517e926ef687c3df04dcd719820de6/notebooks/03-distance-groups-percentages-halved-delay.ipynb

File: requirements.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (leonardoarcari) [reply requirements.txt]: Worderful! Thanks!

**Comment by leonardoarcari** [comment]
Thanks a lot for the fast update! :smile: Job well done!

**Pull request merged**